corehq/ex-submodules/casexml/apps/phone/models.py

```diff
@@ -319,25 +321,18 @@ def update_phone_lists(self, xform, case_list):
         """
         raise NotImplementedError()
 
-    def get_payload_attachment_name(self, version):
-        return 'restore_payload_{version}.xml'.format(version=version)
-
-    def has_cached_payload(self, version):
-        return self.get_payload_attachment_name(version) in self._doc.get('_attachments', {})
+    def set_cached_payload(self, payload_path, version):
+        self.cache_payload_paths[version] = payload_path
 
     def get_cached_payload(self, version, stream=False):
-        try:
-            return self.fetch_attachment(self.get_payload_attachment_name(version), stream=stream)
-        except ResourceNotFound:
-            return None
-
-    def set_cached_payload(self, payload, version):
-        self.put_attachment(payload, name=self.get_payload_attachment_name(version),
-                            content_type='text/xml')
+        response_class = get_restore_response_class(self.domain)
+        if version in self.cache_payload_paths:
+            return response_class.get_payload(self.cache_payload_paths[version])
+        return None
 
     def invalidate_cached_payloads(self):
-        for name in copy(self._doc.get('_attachments', {})):
-            self.delete_attachment(name)
+        self.cache_payload_paths = {}
+        self.save()
```

Comment:
i know this is against our pattern. right now a lot of the code assumes this change impacts the db (see previous implementation). might refactor to only save after